src/main/java/hexgui/util/Options.java

```diff
@@ -0,0 +1,294 @@
+    String value = get(option, Integer.toString(defaultValue));
+    if (value == null) return defaultValue;
+    try {
+      return Integer.parseInt(value);
+    } catch (NumberFormatException e) {
+      throw new Exception("Option -" + option + " needs integer value");
+    }
+  }
+
+  /**
+   * Parse integer option with range check.
+   *
+   * @param option The option key.
+   * @param defaultValue The default value.
+   * @param min The minimum valid value.
+   * @return The option value or defaultValue, if option is not present.
+   * @throws Exception If option value is less than min.
+   */
+  public int getInteger(String option, int defaultValue, int min) throws Exception {
+    int value = getInteger(option, defaultValue);
+    if (value < min) throw new Exception("Option -" + option + " must be greater than " + min);
+    return value;
+  }
+
+  /**
+   * Parse integer option with range check.
+   *
+   * @param option The option key.
+   * @param defaultValue The default value.
+   * @param min The minimum valid value.
+   * @param max The maximum valid value.
+   * @return The option value or defaultValue, if option is not present.
+   * @throws Exception If option value is less than min or greater than max.
+   */
+  public int getInteger(String option, int defaultValue, int min, int max) throws Exception {
+    int value = getInteger(option, defaultValue);
+    if (value < min || value > max)
+      throw new Exception("Option -" + option + " must be in [" + min + ".." + max + "]");
+    return value;
+  }
+
+  /**
+   * Parse long integer option.
+   *
+   * @param option The option key.
+   * @return The option value or 0, if option is not present.
+   * @throws Exception If option value is not a long integer.
+   */
+  public long getLong(String option) throws Exception {
+    return getLong(option, 0L);
+  }
+
+  /**
+   * Parse long integer option.
+   *
+   * @param option The option key.
+   * @param defaultValue The default value.
+   * @return The option value or defaultValue, if option is not present.
+   * @throws Exception If option value is not a long integer.
+   */
+  public long getLong(String option, long defaultValue) throws Exception {
+    String value = get(option, Long.toString(defaultValue));
+    if (value == null) return defaultValue;
+    try {
+      return Long.parseLong(value);
+    } catch (NumberFormatException e) {
+      throw new Exception("Option -" + option + " needs long integer value");
+    }
+  }
+
+  /**
+   * Read options from a file given with the option "config". Requires that "config" is an allowed
+   * option.
+   *
+   * @throws Exception If options in file are not valid according to the specification.
+   */
+  public void handleConfigOption() throws Exception {
+    if (!contains("config")) return;
+    String filename = get("config");
+    InputStream inputStream;
+    try {
+      inputStream = new FileInputStream(filename);
+    } catch (FileNotFoundException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)